WcaOnRails/app/models/competition.rb

```diff
@@ -466,22 +466,76 @@ def country
   end
 
   def self.search(query, params: {})
-    sql_query = "%#{query}%"
-    Competition.where("id LIKE :sql_query OR name LIKE :sql_query OR cellName LIKE :sql_query OR cityName LIKE :sql_query OR countryId LIKE :sql_query", sql_query: sql_query).order(year: :desc, month: :desc, day: :desc)
+    competitions = Competition.where(showAtAll: true)
```

Comment:
Yeah, there isn't (much) less code, but to me it feels cleaner to handle this as separate data, rather than as properties of a half built query object.

You can imagine that would simplify building complex queries. You can also imagine it's just my personal OCD.

I looked around dry-types for a while, but could find anything describing what it actually *does*.